app/main/views/register.py

```diff
@@ -146,6 +146,26 @@ def check_invited_user_email_address_matches_expected(
         debug_msg("invited user email did not match expected email, abort(403)")
         flash("You cannot accept an invite for another person.")
         abort(403)
+    check_for_gov_email_address(user_email)
+
+
+def check_for_gov_email_address(user_email):
```

Comment:
Let's move this method into the [`apps/utils/user.py` file](https://github.com/GSA/notifications-admin/blob/main/app/utils/user.py) and repurpose the [`is_gov_user` method](https://github.com/GSA/notifications-admin/blob/main/app/utils/user.py#L51-L54) to be this check instead.

This would mean a few things:

- The `else` case of flashing a message and aborting with a `403` may still have to live in these views if the method returns false, because the method is used elsewhere in the app and currently returns `True` or `False`.
- Changing the `is_gov_user` method would also make sure that the check mirrors our actual expectations for a government-only user, thus making the other parts of the app that call that method accurate and eliminate potential bugs we haven't caught yet (or at least expose them).
- This also means we could clean up the `apps/utils/user.py` file - anything after that method (so line 56 to the end in its current state) can just be deleted since none of it will be used now.

It also means making an adjustment in the app config, specifically the [`GOVERNMENT_EMAIL_DOMAIN_NAMES` config variable](https://github.com/GSA/notifications-admin/blob/main/app/config.py#L41).  Let's add the `mil` and `si.edu` domains to that list and use the list to check against instead (e.g., `email_address_domain.lower() in config.Config.GOVERNMENT_EMAIL_DOMAIN_NAMES` or something like that).  I'm fine if we keep it just the domain itself or include the preceding `.` with them, whatever makes sense to you.